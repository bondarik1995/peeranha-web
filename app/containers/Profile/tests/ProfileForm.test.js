import React from 'react';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { reduxForm } from 'redux-form';

import ProfileForm from '../ProfileForm';

jest.mock('react-dom');

const Form = reduxForm({
  form: 'ProfileForm',
})(ProfileForm);

const profileMap = new Map();
profileMap.set('profile', {
  ipfs: {},
});

const storeMap = new Map();
storeMap.set('profile', profileMap);

const formFieldValues = {
  handleSubmit: () => {},
  sendProps: {
    profile: {
      ipfs: {},
    },
    translations: {},
  },
};

it('ProfileForm renders correctly', () => {
  const store = createStore(state => state, storeMap);
  const tree = renderer
    .create(
      <Provider store={store}>
        <Form {...formFieldValues} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
