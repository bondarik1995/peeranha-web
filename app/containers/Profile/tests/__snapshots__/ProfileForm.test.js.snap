// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileForm renders correctly 1`] = `
<form>
  <div>
    <div>
      <h6
        className="text-danger"
      />
    </div>
    <div>
      <h6 />
      <input
        className="form-control"
        name="app/Profile/DISPLAY_NAME_FIELD"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        readOnly={true}
        type="text"
        value=""
      />
      <h6
        className="text-danger"
      />
    </div>
    <div>
      <h6 />
      <input
        className="form-control"
        name="app/Profile/POSITION_FIELD"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        readOnly={true}
        type="text"
        value=""
      />
      <h6
        className="text-danger"
      />
    </div>
    <div>
      <h6 />
      <input
        className="form-control"
        name="app/Profile/COMPANY_FIELD"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        readOnly={true}
        type="text"
        value=""
      />
      <h6
        className="text-danger"
      />
    </div>
    <div>
      <h6 />
      <textarea
        className="form-control"
        name="app/Profile/ABOUT_FIELD"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        readOnly={true}
        value=""
      />
      <h6
        className="text-danger"
      />
    </div>
    <div>
      <h6 />
      <div
        className="position-relative mb-0"
      >
        <input
          autoComplete="off"
          className="form-control mb-0"
          name="app/Profile/LOCATION_FIELD"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          readOnly={true}
          type="text"
        />
      </div>
    </div>
  </div>
</form>
`;
